packages/backend/src/apps/m365-excel/common/get-transfer-details.ts

```diff
@@ -41,22 +42,28 @@ async function getTransferDetails(
 
   // TODO (mal/WL): query takes around 0.5s, build cache if necessary
   const tenant = getM365TenantInfo(authData.tenantKey)
-  const results = await $.http.get<{ name: string }>(
-    `/v1.0/sites/${tenant.sharePointSiteId}/drive/items/${fileId}?$select=name`,
-  )
-
-  if (!results.data.name) {
+  try {
+    const results = await $.http.get<{ name: string }>(
+      `/v1.0/sites/${tenant.sharePointSiteId}/drive/items/${fileId}?$select=name`,
+    )
     return {
       position: $.step.position,
       appName: $.app.name,
-      instructions: 'No M365 file is found',
+      connectionName: results.data.name,
     }
-  }
+  } catch (error) {
+    logger.warn('M365 excel file cannot be found after pipe transfer', {
+      event: 'm365-excel-pipe-transfer',
+      error,
+      flowId: $.flow.id,
+      stepId: $.step.id,
+    })
 
-  return {
-    position: $.step.position,
-    appName: $.app.name,
-    connectionName: results.data.name,
+    return {
+      position: $.step.position,
+      appName: $.app.name,
+      instructions: 'M365 file cannot be found in your folder',
```

Comment:
The error should apply only when code === 'itemNotFound'. If not you can return another error message?

```
data: {
   error: {
   code: 'itemNotFound',
   message: 'The resource could not be found.'
   }
},
```

Reply:
ok got it, thanks!